packages/framework/src/Model/Payment/Payment.php

```diff
@@ -119,13 +120,31 @@ public function addTransport(Transport $transport)
     public function setTransports(array $transports)
     {
         $this->transports->clear();
+
         foreach ($transports as $transport) {
             $this->addTransport($transport);
         }
     }
 
     /**
-     * @return \Doctrine\Common\Collections\Collection
+     * @param \Shopsys\FrameworkBundle\Model\Transport\Transport $transport
+     */
+    public function removeTransport(Transport $transport)
+    {
+        $this->transports->removeElement($transport);
+        $transport->removePayment($this);
```

Comment:
This is a circular call. The `Transport` calls also `$payment->removeTransport()` and it probably also fails a line above.

Reply:
Fixed